pkg/git_repo/helpers.go

```diff
@@ -17,3 +21,76 @@ func newHash(s string) (plumbing.Hash, error) {
 	copy(h[:], b)
 	return h, nil
 }
+
+type BasicAuthCredentials struct {
+	Username string
+	Password PasswordSource
+}
+
+type PasswordSource struct {
+	Env        string `yaml:"env"`
+	Src        string `yaml:"src"`
+	PlainValue string `yaml:"value"`
+}
+
+func BasicAuthCredentialsHelper(cfg *BasicAuthCredentials) (*BasicAuth, error) {
+	if cfg == nil {
+		return nil, nil
+	}
+
+	if !oneOrNone([]bool{cfg.Password.Env != "", cfg.Password.Src != "", cfg.Password.PlainValue != ""}) {
+		return nil, fmt.Errorf("password source type could be ONLY `env`, `src` or `value`")
+	}
+
+	password := ""
+
+	if cfg.Password.Env != "" {
+		password = os.Getenv(cfg.Password.Env)
+		if len(password) == 0 {
+			return nil, fmt.Errorf("environment variable %q is not set", cfg.Password.Env)
+		}
+	} else if cfg.Password.Src != "" {
+		absPath, err := util.ExpandPath(cfg.Password.Src)
+		if err != nil {
+			return nil, fmt.Errorf("error load secret from src: %w", err)
+		}
+
+		if exists, _ := util.FileExists(absPath); !exists {
+			return nil, fmt.Errorf("error load secret from src: path %s doesn't exist", absPath)
+		}
+
+		b, err := os.ReadFile(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading secret from file %s: %w", absPath, err)
+		}
+
+		password = string(b)
+	} else if cfg.Password.PlainValue != "" {
+		password = cfg.Password.PlainValue
+	} else {
+		return nil, fmt.Errorf("no secret specified")
+	}
+
+	return &BasicAuth{
+		Username: cfg.Username,
+		Password: password,
+	}, nil
+}
+
+func oneOrNone(conditions []bool) bool {
```

Comment:
I think we could use https://github.com/samber/lo?tab=readme-ov-file#everyby to simplify the code.